### DISCUSSION TITLE
Window showing on linux, but not on windows

### DISCUSSION THREAD

**Discussion by KillTheMule** [fltk-rs/fltk-rs#1029] [Q&A]
I have the following function (I'm leaving out the styling stuff):

```rust
pub fn msg_window(s: &str) {
  let mut wind = Window::default().with_size(
    WIDGET_WIDTH + 4 * PADDING,
    LOGO_SIZE + 9 * BUTTON_HEIGHT + PADDING * 9 + BORDER_DIST * 5, 
  );

  let mut helpdiag = HelpView::default()
    .with_size(
      WIDGET_WIDTH + 2 * PADDING,
      wind.height() - 2 * PADDING - 32 - BUTTON_HEIGHT,
    )
    .with_pos(PADDING, PADDING + 16);

  helpdiag.set_value(s);
  helpdiag.show();

  wind.end();
  wind.make_resizable(true);
  wind.show();
}
```

The problem is, it works nicely on linux, but the helpdialog doesn't shop up when trying on windows. On some tries on windows, it looked a bit like the helpdialog would show up a split second, but I can't really be sure and can't reproduce that. I tried `mem::forget(wind)` to no avail (not surprisingly, this works on linux after all).

Any idea what I could be doing wrong?

Tad of context: The application has a main window, and there's a button which, when clicked, spawns a thread, and this thread is the one that calls `msg_window` in this case. Maybe it's the thread ending that closes the window again? But why doesn't that happen on linux?

Thanks for any pointers!

**Comment by MoAlyousef** [comment]
You can try: 
```rust
while wind.shown() {
    app::wait();
}
```
Generally speaking though, only the main thread should create/destroy windows. It might be that Windows is more strict/sensitive about that than linux. Different behaviors are expected in windowing since FLTK does call the underlying platform's window creation code.

**Reply by KillTheMule** [reply]
Thanks, maybe I'll try that tomorrow, but I'd probably like to take your advice. Would it be possible to create the window in the main thread, and manipulated it from another thread? Or at least just call `show` from another thread? The main thing the "other" thread needs to do is to create the string passed to the helpview, I could easily arrange for it to have a window handle and/or the helpview handle available. I know I could just try, but trying is what made me ask in the first place so... :D

**Reply by MoAlyousef** [reply]
I can’t test things at the moment. But you can try :)
Another option is to use a channel and send a signal to the main thread (with the required string) which would call msg_window

**Reply by KillTheMule** [reply]
Ok, I did, and it seems calling `show` from another thread also does not work on windows. Using signals, things work out, so that's what I'm going to use. I'll add doc comments for `Window::new` and `Window::show` that note this, when I got my app done.
